source/server-discovery-and-monitoring/tests/unified/logging-replicaset.yml

```diff
@@ -0,0 +1,269 @@
+                    ok: 1
+          - level: debug
+            component: topology
+            data:
+              message: "Server heartbeat succeeded"
+              awaited: { $$exists: true }
+              topologyId: { $$exists: true }
+              serverHost: { $$type: string }
+              serverPort: { $$type: [int, long] }
+              driverConnectionId: { $$exists: true }
+              serverConnectionId: { $$exists: true }
+              durationMS: { $$type: [int, long] }
+              reply:
+                $$matchAsDocument:
+                  "$$matchAsRoot":
+                    ok: 1
+          - level: debug
+            component: topology
+            data:
+              message: "Server heartbeat succeeded"
+              awaited: { $$exists: true }
+              topologyId: { $$exists: true }
+              serverHost: { $$type: string }
+              serverPort: { $$type: [int, long] }
+              driverConnectionId: { $$exists: true }
+              serverConnectionId: { $$exists: true }
+              durationMS: { $$type: [int, long] }
+              reply:
+                $$matchAsDocument:
+                  "$$matchAsRoot":
+                    ok: 1
+  - description: Failing hearbeat
+    operations:
+    - name: createEntities
+      object: testRunner
+      arguments:
+        entities:
+        - client:
+            id: *client
+            observeLogMessages:
+              topology: debug
+            observeEvents:
+            - serverHeartbeatStartedEvent
+            - serverHeartbeatFailedEvent
+            uriOptions:
+              heartbeatFrequencyMS: 500
+              appname: failingHeartbeatLoggingTest
+    - name: failPoint
+      object: testRunner
+      arguments:
+        failPoint:
+          configureFailPoint: failCommand
+          mode:
+            times: 2
+          data:
+            failCommands:
+            - hello
+            - isMaster
+            appName: failingHeartbeatLoggingTest
+            closeConnection: true
+        client: setupClient
+    - name: waitForEvent
+      object: testRunner
+      arguments:
+        client: *client
+        event:
+          serverHeartbeatFailedEvent: {}
+        count: 1
+    expectLogMessages:
+    - client: *client
+      ignoreExtraLogs: true
+      ignoreLogMessages:
+        - data:
+            message: "Server heartbeat started"
+        - data:
+            message: "Server heartbeat succeeded"
+        - data:
+            message: "Starting server monitoring"
+        - data:
+            message: "Stopped server monitoring"
+        - data:
+            message: "Topology description changed"
+      messages:
+      - level: debug
+        component: topology
+        data:
+          message: "Starting topology monitoring"
+          topologyId:
+            "$$exists": true
+      - level: debug
+        component: topology
+        data:
+          message: "Server heartbeat failed"
```

Comment:
do we only expect to see one heartbeat failure or should there be one for each host?

Reply:
I think just the primary heartbeat is supposed to fail.

Reply:
is that the case? I'd think the failpoint would apply on any member of the topology. 

that said, I realize now we only set this to times: 2, so it's (in theory) hit once by the RTT monitor and once for a heartbeat, on the first server that we happen to reach out to. 

however, I wonder if it's possible a driver has an interleaving of RTT monitoring commands and server monitoring commands that would lead us to not hit this failpoint on a server monitoring command? i.e. if two RTT monitoring commands hit the failpoint and so it gets disabled before any server monitoring commands encounter it.

it might be more robust to just set the failpoint to always on?